src/functions_framework/__init__.py

```diff
@@ -68,18 +72,54 @@ def view_func(path):
     return view_func
 
 
-def _is_binary_cloud_event(request):
-    return (
-        request.headers.get("ce-type")
-        and request.headers.get("ce-specversion")
-        and request.headers.get("ce-source")
-        and request.headers.get("ce-id")
-    )
+def _run_cloudevent(function, request):
+    data = request.get_data()
+    event = from_http(request.headers, data)
+    function(event)
+
+
+def _cloudevent_view_func_wrapper(function, request):
+    def view_func(path):
+        try:
+            _run_cloudevent(function, request)
+        except cloud_exceptions.MissingRequiredFields as e:
+            flask.abort(
+                400,
+                description=(
+                    "Function was defined with FUNCTION_SIGNATURE_TYPE=cloudevent but"
+                    " failed to find all required cloudevent fields. Found HTTP"
+                    f" headers: {request.headers} and data: {request.get_data()}. "
```

Comment:
Usually request payloads are not logged for several reasons, including:
* Potential for PII or other privileged information
* Larger volume of data than would be useful in log output.

To that end, the various instances of outputting `request.get_data()` should be considered. Was this already happening with the `event` type? Is there a convention in Python where this is expected?

Reply:
After discussin we've decided to move this to a separate PR after this PR